pyccel/parser/semantic.py

```diff
@@ -2139,7 +2139,8 @@ def _visit_FunctionalFor(self, expr, **settings):
         indices = expr.indices
         dims    = []
         body    = expr.loops[1]
-
+        #TODO remove the line below after adding equivalent of atoms to Basic issue #694
+        indices = [sp_Symbol(i.name) for i in indices]
```

Comment:
I think you should name this `sp_indices`. Currently you are creating the final `FunctionalFor` block with sympy objects